docusaurus/docs/learn/core-concepts/security/connection-security.md

```diff
@@ -0,0 +1,46 @@
+# Connection Security
+
+Ziti networks use robust modern cryptography and security mechanisms. Each component within a Ziti network uses
+security technology that fits its role and use-cases.
+
+Here are the different types of connections:
+
+- `control` -  connections between a controller and a router that manage network state. Secured via mTLS 1.2+
+- `link` - connections between routers to form a mesh network. Secured via mTLS 1.2+
+- `edge` - multiplexed connections between an SDK and an Edge Router that carries `service` connections. Secured via mTLS 1.2+
+- `controller APIs` - connections between a controller and an SDK (or other software) via HTTPS and/or secure websockets. Secured via TLS, mTLS 1.2+
+- `service` - connections carried by an `edge` connection and establish communication between a client and host. Application data is end-to-end encrypted via libsodium public private key cryptography.
+
+Below is a diagram showing the `control`, `link`, `edge`, and `controller API` connections. The `service` connections
+exist within an `edge` connection and are pictured in more detail in the second diagram.
+
+![image](/img/connections.png)
+
+
+Connections between SDKs and Edge Routers are called `edge` connections. `edge` connections are multiplexed and carry
+multiple `service` connections. Each connection is for a specific service and secured with end-to-end encryption in 
+order to transport application/service data securely between the two intended parties only.
+
+![image](/img/connections-edge-sdk-sdk.png)
+
+# Connection Walkthrough
```

Comment:
I do think a walkthrough will be helpful, but it feels like there's an opportunity to pace or organize this around the progression of connection types that are being described. It feels like a bunch of dense information that ends a little awkwardly with a definition of multiplexing.

A process sequence might be overkill, but I fed what you wrote into ChatGPT and got this back for us to evaluate as a starting point.

```mermaid
sequenceDiagram
    participant SDK
    participant Controller
    participant Router
    participant EdgeRouter
    participant Service
    
    SDK->>Controller: Authenticate (Edge Client API)
    Controller-->>SDK: API Session security token
    SDK->>EdgeRouter: Dial/bind service (service session security token)
    EdgeRouter-->>SDK: Validate security token, exchange public keys
    SDK->>EdgeRouter: Send data streams (end-to-end encrypted)
    EdgeRouter->>Router: Exchange control & link connections (mTLS)
    Router->>Controller: Coordinate over control connection (mTLS)
```

> In this diagram, the SDK first authenticates with the controller using the Edge Client API and receives an API session security token. Then, the SDK dials or binds a service with the Edge Router, providing the service session security token obtained from the controller's API. The Edge Router validates the security token, exchanges public keys with the client, and establishes end-to-end encrypted data streams for each service connection. Meanwhile, the routers coordinate with the controller over control and link connections that are always authenticated using mTLS.

Reply:
```
sequenceDiagram
    participant SDK
    participant Controller
    participant Router
    participant EdgeRouter
    participant Service
    
    SDK->>Controller: Authenticate (Edge Client API)
    Controller-->>SDK: API Session security token
    SDK->>EdgeRouter: Dial/bind service (service session security token)
    EdgeRouter-->>SDK: Validate security token, exchange public keys
    SDK->>EdgeRouter: Send data streams (end-to-end encrypted)
    EdgeRouter->>Router: Exchange control & link connections (mTLS)
    Router->>Controller: Coordinate over control connection (mTLS)
```

Reply:
If we decide to add a Mermaid.js diagram it can be added in-line with the ` ```mermaid` fencing.

Reply:
I did the same w/ the text, and I didn't much like the chatGPT output - as it removes most of the context a security-oriented reader would want.

I didn't know it could generate mermaid though. Nice.

Reply:
I completely reworked this section. It no longer exists.